src/controllers/redirect.controller.ts

```diff
@@ -28,88 +28,103 @@ export class RedirectController extends BaseController {
+    return this.service.getDissolution(token, session)
   }
 
-  private async handlePendingApprovalRedirect(
-    dissolution: DissolutionGetResponse, dissolutionSession: DissolutionSession, isApplicant: boolean, userEmail: string
-  ): Promise<RedirectResult> {
-    const signingDirector: Optional<DissolutionGetDirector> = this.getUserPendingSignature(dissolution!, userEmail)
+  private async handlePendingPaymentRedirect(dissolution: DissolutionGetResponse, session: DissolutionSession): Promise<RedirectResult> {
+    const userEmail: string = this.session.getUserEmail(this.httpContext.request)
 
-    if (signingDirector) {
-      dissolutionSession.approval = this.prepareApprovalData(dissolution!, signingDirector)
+    const redirectUri: string = this.isApplicant(dissolution, userEmail) ? PAYMENT_URI : CERTIFICATE_SIGNED_URI
 
-      return this.saveSessionAndRedirect(dissolutionSession, ENDORSE_COMPANY_CLOSURE_CERTIFICATE_URI)
-    } else if (isApplicant) {
-      return this.saveSessionAndRedirect(dissolutionSession, WAIT_FOR_OTHERS_TO_SIGN_URI)
-    } else {
-      return this.saveSessionAndRedirect(dissolutionSession, CERTIFICATE_SIGNED_URI)
-    }
+    return this.saveSessionAndRedirect(session, redirectUri)
   }
 
-  private async handlePendingPaymentRedirect(isApplicant: boolean, dissolutionSession: DissolutionSession): Promise<RedirectResult> {
-    if (isApplicant) {
-      return this.saveSessionAndRedirect(dissolutionSession, PAYMENT_URI)
-    } else {
-      return this.saveSessionAndRedirect(dissolutionSession, CERTIFICATE_SIGNED_URI)
+  private async handlePendingApprovalRedirect(dissolution: DissolutionGetResponse, session: DissolutionSession): Promise<RedirectResult> {
+    const userEmail: string = this.session.getUserEmail(this.httpContext.request)
+
+    const signatory: Optional<DissolutionGetDirector> = this.getSignatory(dissolution, userEmail)
+
+    if (signatory && !signatory.approved_at) {
+      session.approval = this.mapper.mapToApprovalModel(dissolution!, signatory)
+      return this.saveSessionAndRedirect(session, ENDORSE_COMPANY_CLOSURE_CERTIFICATE_URI)
+    }
+
+    if (this.isApplicant(dissolution, userEmail)) {
+      return this.saveSessionAndRedirect(session, WAIT_FOR_OTHERS_TO_SIGN_URI)
     }
+
+    return this.saveSessionAndRedirect(session, CERTIFICATE_SIGNED_URI)
   }
 
-  private getUserPendingSignature(dissolution: DissolutionGetResponse, userEmail: string): Optional<DissolutionGetDirector> {
-    return dissolution.directors.find(director => director.email === userEmail && !director.approved_at)
+  private getSignatory(dissolution: DissolutionGetResponse, userEmail: string): Optional<DissolutionGetDirector> {
+    return dissolution.directors.find(director => director.email === userEmail)
   }
 
-  private prepareApprovalData(dissolution: DissolutionGetResponse, signingDirector: DissolutionGetDirector): DissolutionApprovalModel {
```

Comment:
Approval mapping moved out to a separate mapper as all object instantiation should take place within a mapper, as this makes unit testing easier.